File: src/Middleware/integrations/OrderCloud.Integrations.Shipping/Models/ShippingMethod.cs
using System.Collections.Generic;
using Headstart.Common.Services;
using OrderCloud.Integrations.CosmosDB;

namespace OrderCloud.Integrations.Shipping.Models
{
    public class ShippingMethod : CosmosObject
    {
        public string PartitionKey { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public bool Active { get; set; }

        public int? EstimatedTransitDays { get; set; }

        public string Currency { get; set; }

        public TaxCategorization Tax { get; set; }

        public IList<ShippingCost> ShippingCosts { get; set; }

        public string Storefront { get; set; }

        public IList<string> IncludedProductIDs { get; set; }
    }
}

File: src/UI/SDK/src/models/ShippingMethod.ts
import { ShippingCost } from './ShippingCost';
import { TaxCategorization } from './TaxCategorization';

export interface ShippingMethod {
    id: string
    Name?: string
    Description?: string
    Active?: boolean
    EstimatedTransitDays?: number
    Currency?: 'CAD' | 'HKD' | 'ISK' | 'PHP' | 'DKK' | 'HUF' | 'CZK' | 'GBP' | 'RON' | 'SEK' | 'IDR' | 'INR' | 'BRL' | 'RUB' | 'HRK' | 'JPY' | 'THB' | 'CHF' | 'EUR' | 'MYR' | 'BGN' | 'TRY' | 'CNY' | 'NOK' | 'NZD' | 'ZAR' | 'USD' | 'MXN' | 'SGD' | 'AUD' | 'ILS' | 'KRW' | 'PLN'
    Tax?: TaxCategorization
    Storefront?: string
    ShippingCosts?: ShippingCost[]
    IncludedProductIDs?: string[]
    ExcludedProductIDs?: string[]
}